### ISSUE TITLE
control over relationMappings joined by a jsonb field

### ISSUE THREAD

**Issue by MichaelBitard** [Vincit/objection.js#2336]
Hi,

TLDR: this:`ref('titres.propsTitreEtapesIds:points').castText(),` translates to `cast("titres"."props_titre_etapes_ids"#>>'{points}' as text)` and I'd rather have it translate to  `cast("titres"."props_titre_etapes_ids"->>'points' as text)`. Is there a way to do it?

I noticed some slow queries in one of our projects using objections.
After playing with postgres awesome `explain analyze`, I was able to narrow it down to one particular point.

In the generated request, there is this line : ` = cast("titres"."props_titre_etapes_ids"#>>'{points}' as text)` that should not be a search in the jsonb, but directly a reference to it, like that: `= cast("titres"."props_titre_etapes_ids"->>'points' as text)`

This comes from a relationMappings in one of our model: 
```
    communes: {
      relation: Model.ManyToManyRelation,
      modelClass: Communes,
      join: {
        from: ref('titres.propsTitreEtapesIds:points').castText(),
        through: {
          from: 'titresCommunes.titreEtapeId',
          to: 'titresCommunes.communeId',
          extra: ['surface']
        },
        to: 'communes.id'
      }
    },
```

Is there a way to write this differently? I was not able to find this in the documentation around https://vincit.github.io/objection.js/api/types/#type-fieldexpression or https://vincit.github.io/objection.js/api/objection/#ref
